lib/application/widgets/theme/rich_text_field.dart

```diff
@@ -0,0 +1,436 @@
+
+  /// Controls the text content being edited.
+  final RichTextFieldController? controller;
+
+  /// Controls the focus from the content being edited.
+  final FocusNode? focus;
+
+  /// {@macro CustomTextField.errorText}
+  final String? errorText;
+
+  /// {@macro CustomTextField.helperText}
+  final String? helperText;
+
+  @override
+  Widget build(BuildContext context, WidgetRef ref) {
+    final theme = ref.watch(themeController);
+    final textTheme = Theme.of(context).textTheme;
+    final inputDecorationTheme = Theme.of(context).inputDecorationTheme;
+
+    final quillController = _useQuillController(textController: controller);
+
+    final hasContent = useState(controller?.richText.isNotEmpty ?? false);
+
+    useEffect(() {
+      void editorChanged() {
+        final plainText = quillController.plainTextEditingValue.text.trim();
+
+        controller?.richText = jsonEncode(quillController.document.toDelta().toJson());
+        controller?.plainText = plainText;
+        controller?.selection = quillController.selection;
+
+        hasContent.value = plainText.isNotEmpty;
+      }
+
+      quillController.addListener(editorChanged);
+      return () => quillController.removeListener(editorChanged);
+    });
+
+    Future<void> showRichTextFieldModal() async {
+      await showSnappableDraggableModalBottomSheet<void>(
+        context,
+        ref,
+        child: _RichTextFieldModal(
+          title: modalTitle,
+          controller: quillController,
+          placeholder: placeholder,
+          focus: focus,
+        ),
+      );
+    }
+
+    final collapsedEditor = Container(
+      constraints: dimens.richTextFieldConstraints,
+      decoration: BoxDecoration(
+        borderRadius: dimens.genericRoundedElementBorderRadius,
+        color: inputDecorationTheme.fillColor,
+        border: Border.all(
+          color: errorText != null ? theme.destructiveSwatch : inputDecorationTheme.fillColor!,
+          width: dimens.genericBorderHeight,
+        ),
+      ),
+      child: Column(
+        mainAxisAlignment: MainAxisAlignment.center,
+        crossAxisAlignment: CrossAxisAlignment.stretch,
+        mainAxisSize: MainAxisSize.min,
+        children: [
+          if (hasContent.value) ...[
+            Text(placeholder, style: textTheme.caption?.copyWith(color: theme.neutralSwatch.shade400)),
+            context.verticalBox(Spacing.small),
+            Flexible(
+              child: AbsorbPointer(
+                child: _ThemedEditor(
+                  controller: quillController,
+                  placeholder: placeholder,
+                  backgroundColor: theme.neutralSwatch.shade700,
+                  showCursor: false,
+                  readonly: true,
+                ),
+              ),
+            ),
+          ] else
+            Text(placeholder, style: textTheme.subtitle1)
+        ],
+      ).withSymmetricalPadding(context, vertical: Spacing.small, horizontal: Spacing.medium),
+    );
```

Comment:
This looks to me like its own widget. Wdyt?

Reply:
I think yes